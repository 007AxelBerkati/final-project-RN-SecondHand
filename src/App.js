import CodePush from 'react-native-code-push';
import FlashMessage from 'react-native-flash-message';
import { Provider, useSelector, useDispatch } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';

import React, { useEffect } from 'react';
import Router from '@router';

import { LogBox, StatusBar } from 'react-native';
import {
  getNotifikasi, Persistore, Store,
} from './redux';
import { Loading } from './components';

const CodePushOptions = {
  checkFrequency: CodePush.CheckFrequency.ON_APP_START,
  mandatoryInstallMode: CodePush.InstallMode.IMMEDIATE,
  updateDialog: {
    appendReleaseDescription: true,
  },
};

function MainApp() {
  const stateGlobal = useSelector((state) => state.dataGlobal);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getNotifikasi());
  }, []);

  return (
    <>
      <StatusBar barStyle="dark-content" />
      <Router />
      <FlashMessage position="top" />
      {stateGlobal.isLoading && <Loading />}
    </>
  );
}

function App() {
  useEffect(() => {
    LogBox.ignoreLogs(['VirtualizedLists should never be nested']);
  }, []);
  return (
    <Provider store={Store}>
      <PersistGate loading={null} persistor={Persistore}>
        <MainApp />
      </PersistGate>
    </Provider>
  );
}

export default CodePush(CodePushOptions)(App);
