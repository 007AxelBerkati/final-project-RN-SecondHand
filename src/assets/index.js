export * from './icons';
export * from './images';
export * from './dummyData';
