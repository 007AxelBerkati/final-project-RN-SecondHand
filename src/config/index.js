// eslint-disable-next-line import/no-cycle
import instance from './instance';

export { instance };
