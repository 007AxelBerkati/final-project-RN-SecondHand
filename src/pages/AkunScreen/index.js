import {
  ScrollView, StyleSheet, Text, View,
} from 'react-native';
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Fade, Placeholder, PlaceholderMedia } from 'rn-placeholder';
import {
  CardList, Headers, NotLogin, Profile,
} from '../../components';
import {
  borderRadius,
  colors, fonts, fontSize, windowHeight, windowWidth,
} from '../../utils';
import { version } from '../../../package.json';
import { getAkun, logout } from '../../redux';

function AkunScreen({ navigation }) {
  const dispatch = useDispatch();
  const dataProfile = useSelector((state) => state.dataProfile);
  const dataLogin = useSelector((state) => state.dataLogin);

  useEffect(() => {
    dispatch(getAkun());
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const onLogout = () => {
    dispatch(logout());
    navigation.replace('MainApp');
  };

  return (
    <View style={styles.pages}>
      <Headers title="Akun Saya" />
      {
        !dataLogin.isLoggedIn ? (
          <NotLogin onPress={() => navigation.navigate('LoginScreen')} />
        ) : (
          <ScrollView>
            {
              dataProfile.isLoading ? (
                <Placeholder
                  Animation={Fade}
                  style={styles.photoSection}
                >
                  <PlaceholderMedia style={styles.placeholder} />
                </Placeholder>
              ) : (
                <Profile source={{ uri: dataProfile.profile?.image_url }} />
              )
            }
            <CardList type="account" name="edit" title="Ubah Akun" onPress={() => navigation.navigate('ProfileScreen')} />
            <CardList type="account" name="setting" title="Pengaturan Akun" onPress={() => navigation.navigate('PengaturanScreen')} />
            <CardList type="account" name="logout" title="Keluar" onPress={onLogout} />
            <Text style={styles.version}>
              Version
              {' '}
              {version}
            </Text>
          </ScrollView>
        )

      }
    </View>
  );
}

export default AkunScreen;

const styles = StyleSheet.create({
  pages: {
    flex: 1,
    margin: 16,
  },

  version: {
    fontFamily: fonts.Poppins.Regular,
    fontSize: fontSize.small,
    color: colors.text.subtitle,
    marginTop: 16,
    alignSelf: 'center',
  },

  placeholder: {
    borderRadius: borderRadius.xlarge,
    width: windowWidth * 0.3,
    height: windowHeight * 0.15,
    alignSelf: 'center',
  },

  photoSection: {
    alignItems: 'center',
    marginVertical: 24,
  },
});
