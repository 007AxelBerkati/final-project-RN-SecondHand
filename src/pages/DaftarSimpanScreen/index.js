import { View } from 'react-native';
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import CardProduct from '../../components/molekuls/CardProduct';
import { Headers } from '../../components';
import { getWishlistBuyer } from '../../redux';

function DaftarSimpanScreen({ navigation }) {
  const dispatch = useDispatch();
  const data = useSelector((state) => state.data);
  useEffect(() => {
    dispatch(getWishlistBuyer);
  });

  return (
    <View style={{ flex: 1, margin: 16 }}>
      <Headers type="back-title" onPress={() => navigation.goBack()} title="Daftar Simpan" />
      {data?.Product?.map((item) => (
        <CardProduct source={item.image_url} harga={item.base_price} name={item.name} icon="bookmark" />
      ))}

    </View>
  );
}

export default DaftarSimpanScreen;
