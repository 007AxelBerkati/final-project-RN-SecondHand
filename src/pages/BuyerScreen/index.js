import {
  SafeAreaView, ScrollView, StyleSheet, View,
  StatusBar, Text,
} from 'react-native';
import React, { useCallback, useMemo, useRef } from 'react';
import { ImageSlider } from 'react-native-image-slider-banner';
import BottomSheetModal from '@gorhom/bottom-sheet';
import {
  ButtonComponent, CardList, CardProduct, Desc, Gap,
} from '../../components';
import { keranjang } from '../../assets';
import {
  colors, windowHeight, windowWidth,
} from '../../utils';

function BuyerScreen() {
  const bottomSheetModalRef = useRef(null);

  // variables
  const snapPoints = useMemo(() => ['25%', '50%'], []);

  // callbacks
  const handlePresentModalPress = useCallback(() => {
    bottomSheetModalRef.current?.present();
  }, []);
  const handleSheetChanges = useCallback((index) => {
    console.log('handleSheetChanges', index);
  }, []);
  return (
    <SafeAreaView style={styles.pages}>
      <ScrollView showsVerticalScrollIndicator>
        <StatusBar backgroundColor="transparent" translucent />
        <View style={styles.imageContainer}>
          <ImageSlider
            data={[
              { img: 'https://img.freepik.com/free-vector/online-shopping-concept-illustration_114360-1084.jpg?w=1060&t=st=1655909482~exp=1655910082~hmac=17398ae027564d6d34625f5b83f8ab39616da2ea74d55677a11eb56ce8eb70bd' },
              { img: 'https://media.istockphoto.com/photos/male-outfit-set-on-white-background-closeup-top-view-picture-id1176614800?k=20&m=1176614800&s=612x612&w=0&h=uM0CkYhsM12_OYWhnO641tXqD3ZCOx61dyGZClkrv6k=' },
              { img: 'https://images.unsplash.com/photo-1479064555552-3ef4979f8908?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=870&q=80' },
            ]}
            autoPlay
            timer={5000}
            closeIconColor={colors.background.primary}
            caroselImageStyle={{ height: windowHeight * 0.4 }}
            indicatorContainerStyle={{ bottom: windowHeight * 0.05 }}
          />
          <View style={styles.btnBackContainer}>
            <ButtonComponent
              type="icon-button"
              label="BackButton"
            />
          </View>
          <View style={styles.productWrapper}>
            <CardProduct name="Jam tangan" jenis="Barang Antik" harga="10000" />
          </View>
        </View>
        <View style={styles.card}>
          <CardList
            source={keranjang}
            name="Nama Penjual"
            kota="palangka"
            type="role"
          />
        </View>
        <View style={styles.title}>
          <Desc label="Deskripsi" desc="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum." />
        </View>
        <Gap height={60} />
      </ScrollView>
      <View style={styles.btnNego}>
        <ButtonComponent title="saya tertarik dan ingin nego" onPress={handlePresentModalPress} />
      </View>
      <BottomSheetModal
        ref={bottomSheetModalRef}
        index={1}
        snapPoints={snapPoints}
        onChange={handleSheetChanges}
      >
        <View style={styles.contentContainer}>
          <Text>Awesome ðŸŽ‰</Text>
        </View>
      </BottomSheetModal>
    </SafeAreaView>
  );
}

export default BuyerScreen;

const styles = StyleSheet.create({
  imageContainer: {
    height: windowHeight * 0.5,
    width: windowWidth,
  },

  btnBackContainer: {
    position: 'absolute',
    top: 44,
    left: 16,
  },

  title: {
    marginTop: -12,
  },

  card: {
    marginHorizontal: 16,
    marginBottom: 30,
    marginTop: -24,
  },

  pages: {
    flex: 1,
    backgroundColor: colors.background.primary,
  },

  btnNego: {
    padding: 16,
    position: 'absolute',
    width: '100%',
    bottom: 1,
  },

  productWrapper: {
    marginHorizontal: 16,
    marginTop: -40,
  },

});
