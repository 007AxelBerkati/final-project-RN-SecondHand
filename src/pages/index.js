import LoginScreen from './LoginScreen';
import RegisterScreen from './RegisterScreen';
import HomeScreen from './HomeScreen';
import DaftarJualScreen from './DaftarJualScreen';
import AkunScreen from './AkunScreen';
import JualScreen from './JualScreen';
import NotifikasiScreen from './NotifikasiScreen';
import SplashScreen from './SplashScreen';
import InfoPenawaranScreen from './InfoPenawaranScreen';

export {
  LoginScreen,
  RegisterScreen, HomeScreen, DaftarJualScreen, AkunScreen, JualScreen, NotifikasiScreen,
  SplashScreen, InfoPenawaranScreen,
};
