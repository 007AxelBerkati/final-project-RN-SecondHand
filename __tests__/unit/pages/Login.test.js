import React from 'react';
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import { LoginScreen } from '../../../src/pages';
import { Store } from '../../../src/redux';

describe('LoginScreen', () => {
  it('should render LoginScreen', () => {
    const tree = render(
      <Provider store={Store}>
        <LoginScreen />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
