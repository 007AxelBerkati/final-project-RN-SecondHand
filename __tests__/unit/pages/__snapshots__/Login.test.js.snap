// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen should render LoginScreen 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "margin": 16,
    }
  }
  testID="login-screen"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 50,
        "marginTop": 40.019999999999996,
        "opacity": 1,
        "width": null,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#112340",
            "fontSize": 24,
          },
          undefined,
          Object {
            "fontFamily": "Feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ÔÑê
    </Text>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  >
    <Text
      style={
        Object {
          "alignSelf": "flex-start",
          "color": "#FF8303",
          "fontFamily": "Poppins-Bold",
          "fontSize": 30,
          "marginTop": 66.7,
        }
      }
    >
      Masuk
    </Text>
    <View
      style={
        Object {
          "height": 66.7,
          "width": 0,
        }
      }
    />
    <View
      style={
        Object {
          "borderRadius": 47.14285714285714,
        }
      }
    >
      <View
        style={
          Object {
            "color": "#112340",
            "fontFamily": "Poppins-Regular",
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#f6f6f6",
                "borderColor": "#B1B7C2",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
          testID="text-input-outline"
        />
        <View>
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 0,
                },
                Object {
                  "minHeight": 64,
                  "paddingTop": 8,
                },
              ]
            }
          >
            <View
              collapsable={false}
              pointerEvents="none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "translateX": 28,
                    },
                  ],
                  "zIndex": 3,
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "#f6f6f6",
                    "bottom": 4,
                    "left": 10,
                    "maxHeight": 2,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                    "width": 12,
                  }
                }
              />
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                style={
                  Object {
                    "backgroundColor": "#f6f6f6",
                    "color": "transparent",
                    "fontFamily": "System",
                    "fontSize": 22,
                    "fontWeight": undefined,
                    "left": 18,
                    "maxWidth": -28,
                    "opacity": 0,
                    "paddingHorizontal": 0,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 37,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                      Object {
                        "scaleY": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Email
              </Text>
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "#FF8303",
                    "fontFamily": "System",
                    "fontSize": 22,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 0,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 36,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Email
              </Text>
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontFamily": "System",
                    "fontSize": 22,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 0,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 36,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Email
              </Text>
            </View>
            <TextInput
              editable={true}
              maxFontSizeMultiplier={1.5}
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeHolder="Email"
              placeholder=""
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              secureTextEntry={false}
              selectionColor="#FF8303"
              style={
                Array [
                  Object {
                    "flexGrow": 1,
                    "margin": 0,
                    "paddingHorizontal": 14,
                    "zIndex": 1,
                  },
                  Object {
                    "height": 56,
                  },
                  Object {
                    "paddingBottom": 0,
                    "paddingTop": 0,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 22,
                    "fontWeight": undefined,
                    "textAlign": "left",
                    "textAlignVertical": "center",
                  },
                  false,
                  Object {
                    "marginLeft": 44,
                    "paddingLeft": 0,
                  },
                ]
              }
              testID="email"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": 24,
                  "justifyContent": "center",
                  "position": "absolute",
                  "width": 24,
                },
                Object {
                  "left": 12,
                  "top": 23,
                },
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    undefined,
                    Object {
                      "margin": 0,
                    },
                  ],
                ]
              }
            >
              <View>
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  Û∞áÆ
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 16,
          "width": 0,
        }
      }
    />
    <View
      style={
        Object {
          "borderRadius": 47.14285714285714,
        }
      }
    >
      <View
        style={
          Object {
            "color": "#112340",
            "fontFamily": "Poppins-Regular",
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#f6f6f6",
                "borderColor": "#B1B7C2",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
          testID="text-input-outline"
        />
        <View>
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 0,
                },
                Object {
                  "minHeight": 64,
                  "paddingTop": 8,
                },
              ]
            }
          >
            <View
              collapsable={false}
              pointerEvents="none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "translateX": 28,
                    },
                  ],
                  "zIndex": 3,
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "#f6f6f6",
                    "bottom": 4,
                    "left": 10,
                    "maxHeight": 2,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                    "width": 12,
                  }
                }
              />
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                style={
                  Object {
                    "backgroundColor": "#f6f6f6",
                    "color": "transparent",
                    "fontFamily": "System",
                    "fontSize": 22,
                    "fontWeight": undefined,
                    "left": 18,
                    "maxWidth": -28,
                    "opacity": 0,
                    "paddingHorizontal": 0,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 37,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                      Object {
                        "scaleY": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Password
              </Text>
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "#FF8303",
                    "fontFamily": "System",
                    "fontSize": 22,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 0,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 36,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Password
              </Text>
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontFamily": "System",
                    "fontSize": 22,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 0,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 36,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Password
              </Text>
            </View>
            <TextInput
              editable={true}
              maxFontSizeMultiplier={1.5}
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder=""
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              secureTextEntry={true}
              selectionColor="#FF8303"
              style={
                Array [
                  Object {
                    "flexGrow": 1,
                    "margin": 0,
                    "paddingHorizontal": 14,
                    "zIndex": 1,
                  },
                  Object {
                    "height": 56,
                  },
                  Object {
                    "paddingBottom": 0,
                    "paddingTop": 0,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 22,
                    "fontWeight": undefined,
                    "textAlign": "left",
                    "textAlignVertical": "center",
                  },
                  false,
                  Object {
                    "marginLeft": 44,
                    "marginRight": 44,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                  },
                ]
              }
              testID="password"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": 24,
                  "justifyContent": "center",
                  "position": "absolute",
                  "width": 24,
                },
                Object {
                  "left": 12,
                  "top": 23,
                },
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    undefined,
                    Object {
                      "margin": 0,
                    },
                  ],
                ]
              }
            >
              <View>
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  Û∞åÜ
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": 24,
                  "justifyContent": "center",
                  "position": "absolute",
                  "width": 24,
                },
                Object {
                  "right": 12,
                  "top": 23,
                },
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    undefined,
                    Object {
                      "margin": 0,
                    },
                  ],
                ]
              }
            >
              <View>
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#E06379",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  Û∞àâ
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginTop": 16,
        }
      }
    >
      <View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 10.8,
              "elevation": 6,
              "height": 54,
              "justifyContent": "center",
              "opacity": 1,
              "width": 54,
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "borderRadius": 12.571428571428571,
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#FF8303",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              Ó§ç
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 24,
          "width": 0,
        }
      }
    />
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#EDEEF0",
          "borderRadius": 18.857142857142858,
          "opacity": 1,
          "paddingVertical": 10,
        }
      }
      testID="login"
    >
      <Text
        style={
          Object {
            "color": "#B1B7C2",
            "fontFamily": "Poppins-Regular",
            "fontSize": 25.142857142857142,
            "textAlign": "center",
          }
        }
      >
        Login
      </Text>
    </View>
  </RCTSafeAreaView>
  <View
    style={
      Object {
        "height": 133.4,
        "width": 0,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#112340",
          "fontFamily": "Poppins-Medium",
          "fontSize": 22,
        }
      }
    >
      Belum Punya Akun ?
       
    </Text>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FF8303",
            "fontFamily": "Poppins-Bold",
            "fontSize": 22,
            "textAlign": undefined,
          }
        }
      >
        Register
      </Text>
    </View>
  </View>
</View>
`;
